src/v/storage/record_batch_builder.h

```diff
@@ -51,6 +51,8 @@ class record_batch_builder {
      */
     bool empty() const { return _records.empty(); }
 
+    size_t size_bytes() const { return _records.size_bytes(); }
```

Comment:
That is true, however we are just trying to measure the total memory usage within the kafka::clients produce batcher. Not how much it will be when `build()` is called.

Reply:
then perhaps the caller should track how much data it sent into `add_raw_kv`? in any case, adding a doc string would be useful because the api is not intuitive.

Reply:
good point, agreed

Reply:
actually this commit is unused, deadcode, dropping